### 2. В результате эксперимента получены данные приведенные в таблице 2

In [220]:
alpha_1 = 0.02
c = 0.00
d = 1.80
lambda_0 = 0.33
lambda_1 = 1.00
h = 0.4
array = [1.040, 0.802, 0.515, 0.432, 1.052, 0.130, 3.558, 1.081, 2.003, 0.250,
         0.135, 0.406, 1.316, 0.128, 1.470, 2.091, 0.007, 4.259, 2.364, 0.529,
         0.062, 1.139, 0.218, 0.304, 1.942, 0.009, 1.480, 1.867, 0.636, 0.839,
         0.021, 0.469, 1.147, 0.070, 1.476, 1.065, 0.009, 1.532, 0.128, 3.844,
         0.266, 0.744, 1.690, 2.125, 0.310, 0.288, 2.565, 0.334, 2.520, 0.272]

#### a) Построить вариационный ряд. эмпирическую функцию распределения и гистограмму частот.

Вариационный ряд

In [221]:
array.sort()
print(*array, sep=", ")

0.007, 0.009, 0.009, 0.021, 0.062, 0.07, 0.128, 0.128, 0.13, 0.135, 0.218, 0.25, 0.266, 0.272, 0.288, 0.304, 0.31, 0.334, 0.406, 0.432, 0.469, 0.515, 0.529, 0.636, 0.744, 0.802, 0.839, 1.04, 1.052, 1.065, 1.081, 1.139, 1.147, 1.316, 1.47, 1.476, 1.48, 1.532, 1.69, 1.867, 1.942, 2.003, 2.091, 2.125, 2.364, 2.52, 2.565, 3.558, 3.844, 4.259


Эмпирическая функция

In [222]:
dic = {}
f = []
for i in array:
    if i not in dic.keys():
        dic[i] = 1
        f.append(i)
    else:
        dic[i] += 1

cur = 0
print(f"{cur}, x in (-inf, {f[0]}]")
for i in range(len(f)-1):
    cur += dic[f[i]]/len(array)
    print(f"{round(cur, 2)}, x in ({f[i]}, {f[i+1]}]")
print(f"1, x in ({f[len(f)-1]}, inf)")

0, x in (-inf, 0.007]
0.02, x in (0.007, 0.009]
0.06, x in (0.009, 0.021]
0.08, x in (0.021, 0.062]
0.1, x in (0.062, 0.07]
0.12, x in (0.07, 0.128]
0.16, x in (0.128, 0.13]
0.18, x in (0.13, 0.135]
0.2, x in (0.135, 0.218]
0.22, x in (0.218, 0.25]
0.24, x in (0.25, 0.266]
0.26, x in (0.266, 0.272]
0.28, x in (0.272, 0.288]
0.3, x in (0.288, 0.304]
0.32, x in (0.304, 0.31]
0.34, x in (0.31, 0.334]
0.36, x in (0.334, 0.406]
0.38, x in (0.406, 0.432]
0.4, x in (0.432, 0.469]
0.42, x in (0.469, 0.515]
0.44, x in (0.515, 0.529]
0.46, x in (0.529, 0.636]
0.48, x in (0.636, 0.744]
0.5, x in (0.744, 0.802]
0.52, x in (0.802, 0.839]
0.54, x in (0.839, 1.04]
0.56, x in (1.04, 1.052]
0.58, x in (1.052, 1.065]
0.6, x in (1.065, 1.081]
0.62, x in (1.081, 1.139]
0.64, x in (1.139, 1.147]
0.66, x in (1.147, 1.316]
0.68, x in (1.316, 1.47]
0.7, x in (1.47, 1.476]
0.72, x in (1.476, 1.48]
0.74, x in (1.48, 1.532]
0.76, x in (1.532, 1.69]
0.78, x in (1.69, 1.867]
0.8, x in (1.867, 1.942]
0.82, x in (1.

Гистрограмма частот
h = 0.4, n = 50

In [223]:
cur = 0
for i in range(10):
    print(f"H([{round(cur, 1)}, {round(cur + h, 1)})) = {len([i for i in array if cur <= i <= cur + h])/(h*len(array))}")
    cur += h

H([0, 0.4)) = 0.9
H([0.4, 0.8)) = 0.35
H([0.8, 1.2)) = 0.4
H([1.2, 1.6)) = 0.25
H([1.6, 2.0)) = 0.15
H([2.0, 2.4)) = 0.2
H([2.4, 2.8)) = 0.1
H([2.8, 3.2)) = 0.0
H([3.2, 3.6)) = 0.05
H([3.6, 4.0)) = 0.05


#### b) Вычислить выборочные аналоги следующих числовых характер истик: (i) математического ожидания, (ii) дисперсии, (iii) медианы, (iv) асимметрии, (v) эксцесса, (vi) вероятности Pr($X \in [a, b]$).

Вычислим выборочный аналог мат. ожидания

In [224]:
sum = 0

for i in array:
    sum += i

average_x = sum/len(array)
print("average =", round(average_x, 4))

average = 1.0588


$\bar x = \frac{1}{n} \sum_{i=1}^{n}(x_i)=1.0588$

Вычислим выборочный аналог дисперисии

In [225]:
sum = 0

for i in array:
    sum += (i - average_x)**2

dispersion = sum/len(array)

print("S^2 =", round(sum/len(array), 4))

S^2 = 1.0581


$S^2 = \frac{1}{n} \sum_{i=1}^{n}( (x_i - \bar x)^2 )=1.0581$

Вычислим медиану

так как размер выборки четный, то медианой будет среднее значение двух центральных элементов в отсортированном ряде

$M_x = \cfrac{0.744 + 0.802}{2} = 0.773$

In [226]:
print((array[24]+array[25])/2)

0.773


Коэффициент асимметрии

$\Alpha = \frac{\mu_3}{(S^2)^3}\\
\mu = \frac{1}{n} \sum_{i=1}^{n}( (x_i - \bar x)^3 )
$


In [227]:
def make_mu(t):
    mu = 0

    for i in array:
        mu += (i - average_x)**t

    mu /= len(array)

    return mu

In [228]:
A = make_mu(3)/(dispersion**(3/2))

print("A =", round(A, 4))

A = 1.2677


Коэффициент эксцесса
$
\varepsilon = \frac{\mu_4}{(S^2)^4} - 3\\
$

In [229]:
eps = make_mu(4)/(dispersion**(4/2)) - 3

print("Eps =", round(eps, 4))

Eps = 1.1913


Найдем вероятность $Pr(x \in [a, b])$

$Pr(x \in [0, 1.8])=F(1.8)-F(0)=\frac{39}{50}=0.78$


In [230]:
print(len([i for i in array if i < 1.8])/len(array))

0.78


#### c) В предположении, что исходные наблюдения являются выборкой из распределения Пуассона, построить оценку максимального правдоподобия параметра $\lambda$, а также оценку $\lambda$ по методу моментов.

Функция максимального правдоподобия
$L(\vec{x}, \theta)=\prod_{i=1}^{n}f(x_i, \theta)$

Так как функция показаетльного распределения а.н. случайной величины , то $f(x_i, \theta) = p_\lambda(x_i) = f(x_i, \lambda) = \lambda\cdot e^{-\lambda x_i}$

$L(\vec{x}, \lambda)=\prod_{i=1}^{n}\lambda\cdot e^{-\lambda x_i}$

Прологарифмируем функцию

$LL(\vec{x}, \lambda)=\sum_{i=1}^{n}\ln(\frac{\lambda^{x_i}}{x_i !}e^{-\lambda})= n \cdot \ln(\lambda) - \lambda \cdot \sum_{i=1}^n x_i$

Найдем точку масимума 

$\frac{dLL}{d\lambda} = \cfrac{n}{\lambda} - \sum_{i=1}^n x_i\\
\cfrac{d^2LL}{d\lambda^2}=-n\cdot\lambda^{-2} < 0 \Rightarrow \widehat \lambda= \frac{1}{\bar x} = 0.9445$

Найдем оценку $\lambda$ по методу моменту моментов

$Ex_1 = \int_0^{+\infty}x \cdot \lambda\cdot e^{-\lambda x_i} = \frac{1}{\lambda} = \bar x\\
\widehat{\lambda} = \frac{1}{\bar x} = 0.9445$

In [231]:
print(round(1/average_x, 4))

0.9445


#### d) Построить асимптотический доверительный интервал уровня значимости $\alpha_1$ для параметра $\lambda$ на базе оценки максимального правдоподобия.

Найдем функцию $G(x, \lambda)$ - генератор доверительного интервала. $Ex_1= \lambda, \; Dx_1=\frac{1}{\lambda^2} \Rightarrow$

$\sqrt{n} \cdot \frac{\bar x - \lambda}{\sqrt{\lambda}}$, так как $\widehat \lambda=\bar x \Rightarrow$

$\sqrt{n} \cdot \frac{\bar x - \lambda}{\sqrt{\bar x}}$

$P_\theta(x_{1\alpha_1} \le \sqrt{n} \cdot \frac{\bar x - \lambda}{\sqrt{\bar x}} \le x_{2\alpha_1}) \rightarrow 1 - \alpha_1$

Найдем $x_{1\alpha_1}$ и $x_{2\alpha_1}$

$\Phi(x_{\alpha_1}) = 1-\cfrac{\alpha_1}{2} = 1-\cfrac{0.01}{2} = 0.995$, по таблице распределения Пуассона $x_{\alpha_1}=2.81 \Rightarrow$

$P_\theta(-2.81 \le \sqrt{n} \cdot \frac{\bar x - \lambda}{\sqrt{\bar x}} \le 2.81)$, тогда получим интервал

$[-2,81\frac{\sqrt{\bar x}}{\sqrt{n}}-\bar x; \; 2,81\frac{\sqrt{\bar x}}{\sqrt{n}}-\bar x]$

$[\bar x-2,81\frac{\sqrt{\bar x}}{\sqrt{n}}; \; \bar x + 2,81\frac{\sqrt{\bar x}}{\sqrt{n}}]$

$[0.5-2,81\frac{\sqrt{0.5}}{\sqrt{50}}; \; 0.5 + 2,81\frac{\sqrt{0.5}}{\sqrt{50}}]$

$[0.219; \; 0.781]$
